src/material/schematics/ng-generate/m3-theme/index.ts

```diff
@@ -0,0 +1,134 @@
+  const palettes: Map<string, Map<number, string>> = new Map();
+  for (const [key, palette] of Object.entries(materialTheme.palettes)) {
+    const paletteKey = key.replace(/([a-z])([A-Z])/g, '$1-$2').toLowerCase();
+    const tones = paletteKey === 'neutral' ? NEUTRAL_HUE_TONES : HUE_TONES;
+    const colorPalette: Map<number, string> = new Map();
+    for (const tone of tones) {
+      const color = hexFromArgb(palette.tone(tone));
+      colorPalette.set(tone, color);
+    }
+    palettes.set(paletteKey, colorPalette);
+  }
+  return palettes;
+}
+
+function getColorPalettesSCSS(colorPalettes: Map<string, Map<number, string>>): string {
+  let scss = '(\n';
+  for (const [variant, palette] of colorPalettes!.entries()) {
+    scss += '  ' + variant + ': (\n';
+    for (const [key, value] of palette.entries()) {
+      scss += '    ' + key + ': ' + value + ',\n';
+    }
+    scss += '  ),\n';
+  }
+  scss += ');';
+  return scss;
+}
+
+export function generateSCSSTheme(
+  colorPalettes: Map<string, Map<number, string>>,
+  themeTypes: string[],
+): string {
+  let scss = [
+    "@use 'sass:map';",
+    "@use '@angular/material-experimental' as matx;",
+    '',
+    '$_palettes: ' + getColorPalettesSCSS(colorPalettes),
+    '',
+    '$_rest: (',
+    '  secondary: map.get($_palettes, secondary),',
+    '  neutral: map.get($_palettes, neutral),',
+    '  neutral-variant: map.get($_palettes,  neutral-variant),',
+    '  error: map.get($_palettes, error),',
+    ');',
+    '$_primary: map.merge(map.get($_palettes, primary), $_rest);',
+    '$_tertiary: map.merge(map.get($_palettes, tertiary), $_rest);',
+    '',
+  ];
+
+  // Note: Call define-theme function here since creating the color tokens
+  // from the palettes is a private function
+  for (const themeType of themeTypes) {
+    scss = scss.concat([
+      '$m3-' + themeType + '-theme: matx.define-theme((',
+      '  color: (',
+      '    theme-type: ' + themeType + ',',
+      '    primary: $_primary,',
+      '    tertiary: $_tertiary,',
+      '  )',
+      '));',
+    ]);
+  }
+  return scss.join('\n');
+}
+
+function createThemeFile(scss: string, tree: Tree, directory?: string) {
+  const filePath = directory ? directory + 'm3-theme.scss' : 'm3-theme.scss';
+  tree.create(filePath, scss);
+}
+
+export default function (options: Schema): Rule {
+  return async (tree: Tree, context: SchematicContext) => {
+    const logger = context.logger;
+    const colorPalettes = getColorPalettes(options.primaryColor);
+
+    if (options.secondaryColor) {
+      colorPalettes.set('secondary', getColorPalettes(options.secondaryColor).get('primary')!);
+    }
+    if (options.tertiaryColor) {
+      colorPalettes.set('tertiary', getColorPalettes(options.tertiaryColor).get('primary')!);
+    }
+    if (options.neutralColor) {
+      colorPalettes.set('neutral', getColorPalettes(options.neutralColor).get('primary')!);
+    }
+
+    if (options.themeTypes.length === 0) {
```

Comment:
Should we require the user to pick a theme type (or both)?

Reply:
Should the user even pick? Should we just generate both the light and dark themes and users can import the ones they use (or delete the one they don't use within their code)

Reply:
Yeah I think it makes sense to let them pick